pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -148,6 +149,11 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, r *http.Requ
 		return nil, err
 	}
 
+	err = f.validateOpenShiftClientIDUniqueKey(ctx, doc.ClientIDKey)
```

Comment:
I don't think this is quite right.  Below there is a check after Create() for `http.StatusPreconditionFailed` - this is how Cosmos indicates a key violation.  Unfortunately AFAICS Cosmos doesn't tell you *which* key was violated.  Currently we hard code to returning CloudErrorCodeInvalidResourceGroup (the only unique key constraint that exists currently), but now I think it'll be necessary to run an additional query after the failure to determine which constraint failed by trying to find the duplicate.

It's a pain but the lookup shouldn't be too expensive as Cosmos indexes all fields for you and you know the partition key.  You'll need another method on db.OpenShiftClusters to do this, which will call QueryAll under the hood.

I don't think there's an alternative but to try to discover the duplicate *after* the Create failed rather than before - anything else is probably racy.